packages/web-components/fast-foundation/src/disclosure/disclosure.spec.md

```diff
@@ -0,0 +1,119 @@
+
+### Use Cases
+
+-   In general, reveal the extra information on button click when the end-user is needed to know/see.
+-   Laurel wanted to buy a new phone and she was looking at one of the websites where all the basic information was shown. She read about a camera feature called Night Mode and there was a section labeled "See the advanced technology that goes into every Night mode shot". She clicked on it and found out how that feature would work. With that extra information on that feature, she was quite confident in her buying.
+
+### Features
+
+-   Enter and Space: activates the disclosure component and toggles the visibility of the disclosure content.
+-   Use `expanded` attribute or `toggle()` to render default expanded
+-   Observe the state when the disclosure is toggled
+-   `show()` and `hide()` as helper methods to toggle the content from outside
+
+### Prior Art/Examples
+
+-   [Shopify Polaris](https://polaris.shopify.com/components/behavior/collapsible)
+-   [Lightning Design](https://www.lightningdesignsystem.com/components/expandable-section/)
+-   [Bootstrap](https://getbootstrap.com/docs/4.3/components/collapse/)
+-   [Material UI](https://material-ui.com/api/collapse/)
+
+---
+
+## Design
+
+The disclosure component can easily be extended for customization, for example adding animation to reveal the content or get the height of extra content slot.
+
+-   Basic component which toggles the extra content and to add animation extra styles can be applied to derived/extended component.
+-   To have further control to the extra content, there are also `contentNode` and `contentHeight` available as getters.
+
+### API
+
+-   _Component Name:_ `fast-disclosure`
+-   _Props/Attrs:_
+    -   `expanded: boolean` - Current state of the disclosure component
+-   _Methods_:
+    -   `show()`: to show the content
+    -   `hide()`: to hide the content
+    -   `toggle()`: to toggle the content
+-   _Events_
+    -   `toggle: CustomEvent` - No custom data.
+
+### Anatomy and Appearance
+
+```html
+<host>
+    <details>
+        <summary>
+            <slot name="title"></slot>
+        </summary>
+        <div>
+            <slot></slot>
+        </div>
+    </details>
+</host>
+```
+
+-   _Slot Names_
+    -   title: invoker title (could look like as a button or hyperlink)
+    -   default: extra content to be placed
+
+---
+
+## Implementation
+
+Invoker as button
+
+```html
+<fast-disclosure>
+    <span slot="title">More about Green Arrow</span>
+    <div>
+        Green Arrow is a fictional superhero who appears in comic books published by DC
+        Comics. Created by Mort Weisinger and designed by George Papp, he first appeared
+        in More Fun Comics #73 in November 1941. His real name is Oliver Jonas Queen, a
+        wealthy businessman and owner of Queen Industries who is also a well-known
+        celebrity in Star City.
+    </div>
+</fast-disclosure>
+```
+
+Invoker as hypertext
```

Comment:
Can you elaborate on the purpose of this or the intent here? This seems purely visual. I would expect that we would have a single implementation due to functionality. Visually, you could have different appearances - is that the intent?

Reply:
Yes, that's the idea. For example, using it for a blog or description where "Read more" would like a link or button.

Reply:
Gotcha, in this case I don't think we need to call it out in the functional spec as part of the implementation. I think for the FAST implementation, if we wanted to support both appearances, we would do that in the `fast-components` implementation rather than `fast-foundation`. If you take a look at anchor, button, badge - those all have different appearances being applied "outside" the foundational control. The idea is that the semantics used for appearance, as well as the variations may vary by design system. Some may have many, some may have none. In your example above, you call out "as-button" or "as-hypertext". We may use appearances which align with existing semantics such as those in button (stealth, justified, etc).

Reply:
Sounds cool. I will add "neutral" | "hypertext" appearances then. 😅

Reply:
@hardikpthv FWIW, I don't know that we need to call that out in the spec explicitly. The only place I might is under "non-goals". Essentially something like:

"Many times disclosures may take different appearances, such as a button or anchor. Inclusion of such attributes should be applied at the design system / component implementation level".

Reply:
Ah, yes, I got you. I will add it under "Non-goals".